src/components/Modals/FiatRamps/fiatRampProviders/onramper.ts

```diff
@@ -0,0 +1,139 @@
+import { getConfig } from "config";
+import { groupBy, head, uniqBy } from "lodash";
+import { logger } from "lib/logger";
+
+import { FiatRampAction, FiatRampAsset } from "../FiatRampsCommon";
+
+const moduleLogger = logger.child({
+  namespace: ["Modals", "FiatRamps", "fiatRampProviders", "OnRamper"],
+});
+
+// Non-exhaustive required types definition. Full reference: https://github.com/onramper/widget/blob/master/package/src/ApiContext/api/types/gateways.ts
+interface OnRamperGatewaysResponse {
+  gateways: IGatewayItem[];
+  icons: TokenIconMap;
+}
+
+type TokenIconMap = {
+  [key: string]: IconGatewaysResponse;
+};
+
+interface Currency {
+  code: string;
+  id: string;
+  network?: string;
+  displayName?: string;
+}
+
+interface IconGatewaysResponse {
+  name: string;
+  icon: string;
+  symbol?: string;
+}
+
+interface IGatewayItem {
+  identifier: string;
+  cryptoCurrencies: Currency[];
+}
+
+export const getOnRamperAssets = async (): Promise<FiatRampAsset[]> => {
+  const data = await (async () => {
+    try {
+      const baseUrl = getConfig().REACT_APP_ONRAMPER_API_URL;
+      const apiKey = getConfig().REACT_APP_ONRAMPER_API_KEY;
+      const { data } = await axios.get<OnRamperGatewaysResponse>(
+        `${baseUrl}gateways?includeIcons=true`,
+        {
+          headers: {
+            Authorization: `Basic ${apiKey}`,
+          },
+        }
+      );
+
+      return data;
+    } catch (e) {
+      moduleLogger.error(e, "Failed to fetch assets");
+    }
+  })();
+
+  if (!data) return [];
+
+  const fiatRampAssets = convertOnRamperDataToFiatRampAsset(data);
+  return fiatRampAssets;
+};
+
+const convertOnRamperDataToFiatRampAsset = (
+  response: OnRamperGatewaysResponse
+): FiatRampAsset[] => {
+  // We only need Transak coin definitions, they have the cleanest naming scheme out of all available providers
+  const groupedByGateway = groupBy(response.gateways, "identifier");
+  const initialCoins = head(groupedByGateway["Transak"])
+    ?.cryptoCurrencies.map((currency) => {
+      console.log(currency);
+      return toFiatRampAsset(currency, response.icons);
+    })
+    .filter((p) => p !== undefined) as FiatRampAsset[];
+
+  const uniqueCoins = uniqBy(initialCoins, "assetId");
+  return uniqueCoins;
+};
+
+function toFiatRampAsset(
+  currency: Currency,
+  icons: TokenIconMap
+): FiatRampAsset | undefined {
+  const assetId = adapters.onRamperTickerToAssetId(currency.code);
+  if (assetId !== undefined) {
+    return {
+      name: currency.displayName || "",
+      assetId,
+      symbol: currency.code,
+      imageUrl: icons[currency.code].icon,
+      fiatRampCoinId: currency.id
+    };
+  }
+  return undefined;
+}
+
+export const createOnRamperUrl = (
+  action: FiatRampAction,
+  assetId: AssetId,
+  address: string
+): string => {
+  const onRamperSymbol = adapters.assetIdToOnRamperTicker(assetId);
+  // get the supported assets for this chain based on assetId
+  // and filter available tokens for this trade
```

Comment:
So by default, onramper let's you trade any pair with the address you've provided. Thankfully we can at least prevent address updates by disabling them, but the user can pick an incompatible coin for a given address and thus loose his cash. Two options here:
- provide a list of "allowed" traders for a given address with each redirect
- always limit the user to the one coin he's selected in our UI

I've already started implementing nr1, but nr2 is much simpler and gets the job done, so I guess we should go with that